### ISSUE TITLE
8.0.0:JJTree:C++: generated code different from the 7.0 version

### ISSUE THREAD

**Issue by zosrothko** [javacc/codegenerator#15]
Hi

The generated code by JJTree/JavaCC 8.0.0 for a C++ generation is diferent from the one by JJTree/JavaCC 7.0.5 for the same grammar, here <javacc>/examples/JJTreeExamples/cpp/eg2.jjt

8.0.0 generates this  ParserTree.h below
```
/* Generated By:JJTree: Do not edit this line. ParserTree.h Version 8.0 */
/* JavaCCOptions:MULTI=true,NODE_USES_PARSER=false,VISITOR=false,TRACK_TOKENS=false,NODE_PREFIX=AST,NODE_EXTENDS=,NODE_FACTORY=,SUPPORT_CLASS_VISIBILITY_PUBLIC=true */
#ifndef PARSERTREE_H
#define PARSERTREE_H
#include <vector>
#include "JavaCC.h"
#include "Token.h"
#include "Node.h"
#include "ParserTreeConstants.h"


namespace EG2 {
class SimpleNode : public Node {

  protected: Node *parent;
  protected: std::vector<Node*> children;
  protected: int id;
  protected: void * value;
  protected: Parser *parser;
  //protected: int numChildren;

  public: SimpleNode(int i);
  public: SimpleNode(Parser *p, int i);

//#if !NODE_FACTORY
//\#define jjtCreate(id) new SimpleNode(id)
//\#define jjtCreate(p, id) new SimpleNode(p, id)
//#fi

  public: virtual void  jjtOpen() const;
  public: virtual void  jjtClose() const;
  public: virtual void  jjtSetParent(Node *n);
  public: virtual Node* jjtGetParent() const;
  public: virtual void  jjtAddChild(Node *n, int i);
  public: virtual Node* jjtGetChild(int i) const;
  public: virtual int   jjtGetNumChildren() const;
  public: virtual void  jjtSetValue(void * value);
  public: virtual void* jjtGetValue() const;

  /* You can override these two methods in subclasses of SimpleNode to
     customize the way the node appears when the tree is dumped.  If
     your output uses more than one line you should override
     toString(string), otherwise overriding toString() is probably all
     you need to do. */

  public: virtual JJString toString() const;
  public: virtual JJString toString(const JJString& prefix) const;

  /* Override this method if you want to customize how the node dumps
     out its children. */

  public: virtual void dump(const JJString& prefix) const;
  public: virtual void dumpToBuffer(const JJString& prefix, const JJString& separator, JJString *sb) const;
  private: virtual std::vector<Node *> jjtExtractChildrenForDestruction() override;
  public: virtual ~SimpleNode();
  public: virtual int getId() const {
    return id;
  }
};
class ASTStart : public SimpleNode {
public: 
           ASTStart(int id);
           ASTStart(Parser *parser, int id);
  virtual ~ASTStart();

};
class ASTMyID : public SimpleNode {
public: 
           ASTMyID(int id);
           ASTMyID(Parser *parser, int id);
  virtual ~ASTMyID();
};
```
while 7.0.5 generates this ParserTree.h below
```
/* Generated By:JJTree: Do not edit this line. ParserTree.h Version 7.0 */
/* JavaCCOptions:MULTI=true,NODE_USES_PARSER=false,VISITOR=false,TRACK_TOKENS=false,NODE_PREFIX=AST,NODE_EXTENDS=,NODE_FACTORY=,SUPPORT_CLASS_VISIBILITY_PUBLIC=true */
#ifndef PARSERTREE_H
#define PARSERTREE_H
#include "SimpleNode.h"
#include "ASTStart.h"
#include "ASTMyID.h"
#include "ASTMult.h"
#include "ASTInteger.h"
#include "ASTAdd.h"
#include "ASTUnaryExpression.h"
#endif
/* JavaCC - OriginalChecksum=82c835150c5f456599281cbebb3b0032 (do not edit this line) */
```

**Comment by mbrigl** [comment]
Fixed with pull request #18

**Comment by zosrothko** [comment]
Verified
